library/Notifications/Hook/EventsObjectsInfoHook.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/* Icinga Notifications Web | (c) 2024 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Notifications\Hook;
+
+use Exception;
+use Icinga\Application\Hook;
+use Icinga\Application\Logger;
+use Icinga\Module\Notifications\Model\Objects;
+use ipl\Html\HtmlString;
+use ipl\Html\ValidHtml;
+use ipl\Web\Url;
+use ipl\Web\Widget\Link;
+
+abstract class EventsObjectsInfoHook
+{
+    /** @var array<string, ValidHtml>  */
+    protected static $objectNameWidgets = [];
+
+    /**
+     * Get the object name widgets for the objects using the object ID tags
+     *
+     * @param array<string, array<string, string>> $objectIdTags
+     *
+     * @return array<string, ValidHtml>
```

Comment:
This type hint seems to be incorrect.
It should be `@return array<string, array<string, ValidHtml>` // Outer array key must be source type, inner array key must be object id. Please document this.